lib/src/util.dart

```diff
@@ -83,9 +84,15 @@ final _oneOrMoreWhitespacePattern = RegExp('[ \n\r\t]+');
 
 /// "Normalizes" a link label, according to the [CommonMark spec].
 ///
-/// [CommonMark spec] https://spec.commonmark.org/0.29/#link-label
-String normalizeLinkLabel(String label) =>
-    label.trim().replaceAll(_oneOrMoreWhitespacePattern, ' ').toLowerCase();
+/// [CommonMark spec] https://spec.commonmark.org/0.30/#link-label
+String normalizeLinkLabel(String label) {
+  final text = label.trim().replaceAll(_oneOrMoreWhitespacePattern, ' ');
+  final buffer = StringBuffer();
+  for (var i = 0; i < text.length; i++) {
```

Comment:
My only concern with this PR is performance. At this line. Two bits:

* For every character, we  have to do these two lookups (`text[]` and `caseFoldingMap[]`)
* Even if `text[i]` is not in `caseFoldingMap`, we still perform an action (add to the StringBuffer).

Can you post some benchmark results from `benchmark/benchmark.dart`, please?

Reply:
### Now
![Screenshot 2022-10-06 at 20 36 16](https://user-images.githubusercontent.com/5637609/194392431-232de8d3-9830-4dc6-b07e-178d13ca56e6.png)

### Before
![Screenshot 2022-10-06 at 20 36 55](https://user-images.githubusercontent.com/5637609/194392448-c0b20224-b2b8-4a18-8bdc-2207f4473f8d.png)